main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1361,7 +1361,7 @@ object Lowering {
         val locksSym = mkLetSym("locks", loc)
         val pat = mkTuplePattern(List(LoweredAst.Pattern.Cst(Ast.Constant.Int32(i), Type.Int32, loc), LoweredAst.Pattern.Var(locksSym, locksType, loc)), loc)
         val getTpe = Type.eraseTopAliases(chan.tpe) match {
-          case Type.Apply(_, t, _) => t
+          case Type.Apply(Type.Apply(_, t, _), _, _) => t
```

Comment:
@magnus-madsen I'm not really happy with this: it feels rather fragile to me. But I'm not sure that there's any better way? Suggestions welcome.

Reply:
This is acceptable. Your other alternative is `.typeArguments.head`, but what you have is fine.